src/generation.rs

```diff
@@ -6869,46 +6877,72 @@ fn generate_enum(
                         };
-                        let (before, after) =
-                            if cli.preserve_encodings || !variant.rust_type().is_fixed_value() {
-                                (Cow::from(format!("let {var_names_str} = ")), ";")
-                            } else {
-                                (Cow::from(""), "")
-                            };
+                        let (before, after) = if cli.preserve_encodings
+                            || !variant.rust_type().is_fixed_value()
+                            || rep.is_some()
+                        {
+                            (Cow::from(format!("let {var_names_str} = ")), ";")
+                        } else {
+                            (Cow::from(""), "")
+                        };
                         let mut variant_deser_code = gen_scope.generate_deserialize(
                             types,
                             (variant.rust_type()).into(),
                             DeserializeBeforeAfter::new(&before, after, false),
                             DeserializeConfig::new(&variant.name_as_var()),
                             cli,
                         );
-                        let names_without_outer = enum_gen_info.names_without_outer();
-                        // we can avoid this ugly block and directly do it as a line possibly
-                        if variant_deser_code.content.as_single_line().is_some()
-                            && names_without_outer.len() == 1
-                        {
-                            variant_deser_code = gen_scope.generate_deserialize(
-                                types,
-                                (variant.rust_type()).into(),
-                                DeserializeBeforeAfter::new(
-                                    &format!("Ok({}::{}(", name, variant.name),
-                                    "))",
-                                    false,
-                                ),
-                                DeserializeConfig::new(&variant.name_as_var()),
-                                cli,
-                            );
-                        } else if names_without_outer.is_empty() {
-                            variant_deser_code
-                                .content
-                                .line(&format!("Ok({}::{})", name, variant.name));
+                        if let Some(r) = rep {
+                            let len_info = match ty.conceptual_type.resolve_alias_shallow() {
+                                ConceptualRustType::Rust(ident) if types.is_plain_group(ident) => {
+                                    types.rust_struct(ident).unwrap().cbor_len_info(types)
+                                }
+                                _ => RustStructCBORLen::Fixed(1),
+                            };
+                            // this will never be 1 line to don't bother with the below cases
```

Comment:
not sure I understand what this comment means. Maybe `to` was meant to be `so`?

Reply:
Yes. Fixed.